Firebase/Core/FIROptions.m

```diff
@@ -346,6 +346,56 @@ - (void)setAppGroupID:(NSString *)appGroupID {
   _appGroupID = [appGroupID copy];
 }
 
+#pragma mark - Equality
+
+- (BOOL)isEqual:(id)object {
+  if (!object || ![object isKindOfClass:[FIROptions class]]) {
+    return NO;
+  }
+
+  return [self isEqualToOptions:(FIROptions *)object];
+}
+
+- (BOOL)isEqualToOptions:(FIROptions *)options {
+  // Skip any non-FIROptions classes.
+  if (![options isKindOfClass:[FIROptions class]]) {
+    return NO;
+  }
+
+  // Check the internal dictionary and custom properties for differences.
+  if (![options.optionsDictionary isEqualToDictionary:self.optionsDictionary]) {
+    return NO;
+  }
+
+  // Validate extra properties not contained in the dictionary. Only validate it if one of the
+  // objects has the property set.
+  if ((options.deepLinkURLScheme != nil || self.deepLinkURLScheme != nil) &&
+      ![options.deepLinkURLScheme isEqualToString:self.deepLinkURLScheme]) {
+    return NO;
+  }
+
+  if ((options.appGroupID != nil || self.appGroupID != nil) &&
+      ![options.appGroupID isEqualToString:self.appGroupID]) {
+    return NO;
+  }
+
+  // Validate the Analytics options haven't changed with the Info.plist.
+  if (![options.analyticsOptionsDictionary isEqualToDictionary:self.analyticsOptionsDictionary]) {
+    return NO;
+  }
+
+  // We don't care about the `editingLocked` or `usingOptionsFromDefaultPlist` properties since
+  // those relate to lifecycle and construction, we only care if the contents of the options
+  // themselves are equal.
+  return YES;
+}
+
+- (NSUInteger)hash {
+  // This is strongly recommended for any object that implements a custom `isEqual:` method to
+  // ensure that dictionary and set behavior matches other `isEqual:` checks.
+  return self.optionsDictionary.hash ^ self.deepLinkURLScheme.hash ^ self.appGroupID.hash;
```

Comment:
nit: Was `analyticsOptionsDictionary` omitted here intentionally?

Reply:
Yes, added a comment mentioning why that was the case. Thanks!

Reply:
Thank you!